### DISCUSSION TITLE
*Solved* Error starting dockercontainer /run/s6/basedir/scripts/rc.init: 67: restart:: not found

### DISCUSSION THREAD

**Discussion by snakekick** [alexjustesen/speedtest-tracker#409] [Q&A] [answered]
Hello, 
i try to use the speedtest-tracker on an arm64 with armbian.
When i start the container i get this error:
/run/s6/basedir/scripts/rc.init: 67: restart:: not found

this is the complete output, thank you:

To support Server Side Up projects visit:
https://serversideup.net/sponsor
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    131
-------------------------------------

🔒 SSL_MODE has been set to MIXED, setting the web server to work in HTTP + HTTPS...
🏃‍♂️ Checking for Laravel automations...
🔐 Linking the storage...

   ERROR  The [public/storage] link already exists.  

🐇  Configuring Speedtest Tracker...
✅  Database exists
🔗  Creating database symlink
✅  Environment file exists
🔗  Creating symlinks to config and log files
✅  App key exists
🔒  Fixing app path file permissions
💰  Building the cache...
[15-Feb-2023 10:52:33] NOTICE: fpm is running, pid 95
[15-Feb-2023 10:52:33] NOTICE: ready to handle connections
[15-Feb-2023 10:52:33] NOTICE: systemd monitor interval set to 10000ms
❌ There seems to be a failure in checking the web server + PHP-FPM. Here's the response:

🚛  Migrating the database...
✅  All set, starting Speedtest Tracker container...
⏰  Starting the cron service...
💪  Starting the queue worker...
/run/s6/basedir/scripts/rc.init: 67: restart:: not found
[15-Feb-2023 10:52:35] NOTICE: Terminating ...
[15-Feb-2023 10:52:35] NOTICE: exiting, bye-bye!

**Comment by nomoreimportant** [comment]
I am getting the same error and unable to get more details. I am running this on a raspberry pi arm64 with the below docker command. I looked at all the configuration details and could not find more details.

docker run -d --name speedtest-tracker \
    -p 4488:80 \
    -p 4483:443 \
    -e PUID=1000 \
    -e PGID=1000 \
    -v /portainer/Files/AppData/Config/speedtest:/config \
    ghcr.io/alexjustesen/speedtest-tracker:latest \
    restart: unless-stopped

I have to provide permission as it was giving an error to create sqlite file.
sudo chown -R 1000:1000 /portainer/Files/AppData/Config/speedtest/

Log details below
--------------------------------------------------------------------
To support Speedtest Tracker visit:
https://github.com/alexjustesen/speedtest-tracker
-------------------------------------
GID/UID
-------------------------------------
User uid:    1000
User gid:    1000
-------------------------------------
🔒 SSL_MODE has been set to MIXED, setting the web server to work in HTTP + HTTPS...
🏃‍♂️ An SSL key was not detected, so I'll generate a self-signed SSL certificate pair for you...
..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+.......+...+..+.......+.....+.+..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.................+......+.+.....+.+..+..........+...+..............+....+..+...+.......+...+......+........+.+......+.........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
.......+.....+.+..+...+.......+..+.........+....+..+.+...+..+....+...+........+....+.........+.....+.+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+........+.......+...+..+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+......+..................+..+....+..+.......+...+...+...........+...+..........+......+..+....+.....+.+..+.+......+.....+....+.................+...............+...+...................+........+...+.......+..+.......+...............+.....+...+...+...+.......+........+....+......+..+.........+......+....+.....+....+......+...+.....+...+.......+..+.+........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-----
🐇  Configuring Speedtest Tracker...
🙄  SQLite database not found, creating...
✅  SQLite database created.
🔗  Creating database symlink...
✅  Database symlink created.
🙄  Environment file not found, creating...
✅  Environment file created.
🔗  Creating symlinks for config and log files...
✅  Symlinks created.
⏳  Generating app key...
✅  App key generated.
🔒  Fixing app path file permissions...
✅  Permissions fixed.
🔐  Linking the storage...
✅  Storage linked.
💰  Building the cache...
✅  Cache set.
🚛  Migrating the database...
✅  Database migrated.
✅  All set, Speedtest Tracker started.
[07-Mar-2023 23:28:05] NOTICE: fpm is running, pid 135
[07-Mar-2023 23:28:05] NOTICE: ready to handle connections
[07-Mar-2023 23:28:05] NOTICE: systemd monitor interval set to 10000ms
/run/s6/basedir/scripts/rc.init: 67: restart:: not found
[07-Mar-2023 23:28:06] NOTICE: Terminating ...
[07-Mar-2023 23:28:06] NOTICE: exiting, bye-bye!

**Comment by hasecilu** [comment, marked as answer]
I also had this problem, the folder was created by user root. Making `sudo chown -R 1000:1000 /folder/path` solved it, thanks! 
```
03/10/2023 12:03:41 PM 🙄  SQLite database not found, creating...
03/10/2023 12:03:41 PM touch: cannot touch '/config/database.sqlite': Permission denied
03/10/2023 12:03:41 PM s6-rc: warning: unable to start service laravel-automations: command exited 1
03/10/2023 12:03:41 PM /run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up!  Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
```

**Reply by Cptnspoon** [reply]
I could be wrong, but I don't believe this is the answer to the original problem.  That being the line 

/run/s6/basedir/scripts/rc.init: 67: restart:: not found

I'm not super knowledgeable about linux and docker, but i finally fixed this issue by using docker compose instead of standard docker, specifically by following along with the video linked in the intro by Awesome Open Source.  Don't know why it worked tbh

**Reply by hasecilu** [reply]
I was reading #430 because when using docker compose couldn't start server with sqlite, also tried shell command but nothing worked.
I had similar symptoms like `/run/s6/basedir/scripts/rc.init: 67: restart:: not found`. The only relevant change that I made was changing owner of the config folder because the owner was root, that worked for me but I'm also a docker newbie.

**Reply by axeloz** [reply]
Hello, 

I doesn't work for me: 
```
docker run -d --name speedtest-tracker \
    -p 8081:80 \
    -e PUID=1001 \
    -e PGID=1001 \
[...]
```
And I did a `sudo chmod -R 1001:1001` but still the same issue. I would rather use Docker instead Docker-compose as I already have a MySQL server running on the host. But it is true that Docker-compose works
Thanks

**Reply by axeloz** [reply]
To solve this, I gave up on using Docker and used Docker compose instead. I just removed the database settings in docker-compose.yml and used my local database instead.
